src/profile/external-identities/ExternalIdentities.tsx

```diff
@@ -12,23 +12,22 @@ type ExternalIdentitiesProps = {
 
 export const ExternalIdentities = (props: ExternalIdentitiesProps): ReactNode => {
   const { queryParams } = props;
+  const providers = getConfig().externalCreds?.providers || [];
+  const hasRas = providers.includes('ras');
+  const filteredProviders = hasRas ? providers.filter((p: string) => p !== 'era-commons') : providers;
```

Comment:
1) instead of including runtime logic to hide era-commons if we have a ras provider … could you remove era-commons from the dev and staging configs where ras is present?

2) The logic to filter providers is split between here and line 23 below. Instead, could you consolidate all the filtering logic up here ("arrange"), then use a simple loop without a filter below ("act")?

I suppose the logic would be something like this … though the logic will need to change if you implement the first suggestion:
```ts
  const filteredProviders = providers.filter(
    (p: string) =>
      (p === 'era-commons' && !hasRas) ||
        (p === 'github' && githubEnabled) ||
        (p !== 'github' && p !== 'era-commons')
  );
```

Reply:
Fixed 1: [c6596ee](https://github.com/DataBiosphere/terra-ui/pull/5259/commits/c6596eede05a55d7c24a316ccdbfab79393b5afb) and 2: https://github.com/DataBiosphere/terra-ui/pull/5259/commits/43155832f6b18800211928eb9acb18bb4d48a860